### DISCUSSION TITLE
Bbshd current 33A

### DISCUSSION THREAD

**Discussion by Paulii6661** [danielnilsson9/bbs-fw#225] [General]
hello,

ive seen on the internet people doing hardware modification for max amps called "shunt mode".

my question is is it possible change the setting in tool to more than 33A?

i understand it can kill my controller. but im really wonderging if firmware can do same as shunt mode or not. and if it can which files do i need to adjust?

**Comment by Energeek136** [comment]
I think shunt mode is done by hardware modification, putting solder onto a shunt resistor and making it lower resistance and tricking the software into thinking its running at lower current. Therefore when lets say 40 amps are going through the shunt the software will only see lets say 30

**Reply by Paulii6661** [reply]
I know I was just wondering how far can you push it without having to do all the hardware work. But actually you can trick the saved config and change it directly there through txt editor but I would strongly advise against it.

**Comment by dav0000000** [comment]
You can change the setting to higher than 33amps but it won't do anything, at least it won't on the BBSHD. This gets set in motor.c with a 0-255 value being sent to a secondary nec controller. The maximum value it will accept on the BBSHD is 220. If you send higher it will set it to 220. That makes me think that the define for ADC_STEPS_PER_AMP_X10 should actually be 220/33 rather than 69 if the true max amps is 33a.

**Reply by Paulii6661** [reply]
Oh come on I was so happy for now obviously placebo effect after setting it to 35A :D I'm really thinking now to do shunt mode but I'm a bit afraid because when you hold full throttle it goes to 15A and then second after to 30A it would be a bit scary if it jumped from 30 to 60A also I don't know if my battery could handle 60A for too long :/

**Comment by WvBijsterveld** [comment]
if your battery can’t handle it your bms will save it and you will know. also i could be very wrong but i think the bursts don’t quite hit that hard.
also as far as i know shunt mod is not soldering directly but instead replace 1 or 2 mosfets so you get a 1.5 multiplier for example. or a 2x but you will still set your target A on 25 to get 50 max for example. bursts are far more easier on the battery and i think it should handle it.

on my stock battery that shipped with my bs02 (used on my bbshd with other controller) i ran 2.9kw continuous with it only getting warm.

**Reply by Paulii6661** [reply]
Finally someone with this mod experience 2.9kw sounds insane for that motor I'm running 33A and after today's 5min full speed full current meaning 1.6kw and going 60km/h I got 65°c motor. Did you have any issues with the motor at that power? Nylon gear or controller failure?

**Comment by WvBijsterveld** [comment]
it wasn’t shunt but just external controller and no, i’m not joking but i runs even smoother on 4/5 kw with another battery then on 1kw.
 just the external controller being able to give the motor so much rpm makes it so i can just have 1 gear and it doesn’t heat up the motor at all it’s so good.

**Reply by Energeek136** [reply]
How much rpm are you running? Is normal rpm around 10000? At least thats what I saw on the internet

**Reply by Paulii6661** [reply]
Oh I want to keep stock controller for the looks I mean I have enough power for wheelies I was thinking to add some power for more speed.

But still I have 20ah battery with 60A peak I would have about 20min of ride time.

**Comment by WvBijsterveld** [comment]
the internal rpm can reach 15k. not sure what that means for chainring rpm. i’m on 72v.

**Comment by WvBijsterveld** [comment]
honestly, don’t bother adding more power. for me it was relatively expensive to go from 50/60 to high eighty.
i still get 50km on full power with my 1400wh battery .
but primary thing to get higher top speed is volts. u are just limited to the max rpm the motor can spin.

i have enough power to auto wheelie with 36t at medium speed even but i find the throttle curve hard to wheelie with.
also not at all good at doing wheelies lol.

**Reply by Paulii6661** [reply]
Well yeah that's why I'm thinking of shunt moding it shut costs nearly nothing.

You need both volts and amperes. You can have 72v battery and motor can spin fast but unless you add enough power then wind will stop you rather quickly so you can go fast with 52v and a lot of power because you can use low gears and push through wind with enough power.

I have to yank (or how to say in English) to get wheelie (42t front and 42 back). Also for some reason when you do full throttle you get I guess 15a and then a second later it jumps 30a sadly Daniel told me it's not firmware issue but probably second controller issue.

How is it with different controller? I noticed I have very high torque until around half max rpm and then torque rather quickly decays.

**Comment by WvBijsterveld** [comment]
yes you’re right, in my experience i felt like i needed the volts more(especially since i was on 48v) then actual power.
my torque is great all round on the rpm range. i can set the field weakening which can increase max rpm at the cost of acceleration.

do you just full throttle it once in wheelie and then brake when necessary, or can you actually keep it going just by playing with the throttle?

oh yea the thing i hated most about the bbshd controller was that the power was not instant, the thing you say with first 15a then 30 is what was a real bummer. on the bbs02 it’s instant, just standing still accidentally press the throttle in all the way and it flies. really sad they made that section worse on the big power model, made my trying to wheelie journey really hard and demotivated me.

**Reply by Paulii6661** [reply]
Well with 52v and ratio 1:1 I can get about 25km/h with 29" wheels. And I'm kinda okay with that doing wheelis faster than that I would be probably scared I'm 30yo so I'm happy to do at least wheelie not a fast wheelie :D

I hold a little throttle and then I go full throttle that seems to go around that 15 - 30A Issue and if necessary I use rear brake to balance (I didn't install rear brake sensor)

Yeah that's a bummer but I kinda get it what manufacturer did you know pushing 30A to the nylon gear when you are at complete stop and for example in the lowest gear would potentially damage a lot of stuff... My wife have tsdz2 750w and all the time she is at 10th gear which has only 11 teeth and in front is 42... I belive if I did it with bbshd there would be very expensive noise.

**Comment by WvBijsterveld** [comment]
hahaha that’s sick be proud of your wheelie, i’m 18 and can’t do it for shit :P

yea i sort of get it too but the bbs02 can run  30a on this firmware aswell (or a little less idk anymore)

when i do it now with this controller when i’m standing still, it cuts out. so wouldn’t be that hard for bafang to implement that instead of ramping up and basically giving up a bit of user precision control.

**Reply by Paulii6661** [reply]
Now I feel even older than 10 minutes ago :D I have basically 0 experience with bbs02 I had tongsheng and went straight for bbshd. And yeah like you are saying but sadly there's not much I can do to change it except for buying external controller and I don't really like that idea at least for now.

**Comment by WvBijsterveld** [comment]
lol yea i first wanted a gasoline engine on my bicycle but then saw that there weren’t really any sets/kits. i then saw the bafang motors and fell in love, didn’t want to spend too much so went with the bbs02, then felt the need to upgrade and go faster, then again with external controller, then now again but i’m saving up so can’t lol.
